src/server/database/Database/DatabaseWorkerPool.cpp

```diff
@@ -377,6 +379,52 @@ void DatabaseWorkerPool<T>::KeepAlive()
         Enqueue(new PingOperation);
 }
 
+// Returns true if the version string given is incompatible
+//
+// Intended to be used with mysql_get_server_info()'s output as the source
+//
+// DatabaseIncompatibleVersion("8.0.35") => false
+// DatabaseIncompatibleVersion("5.6.6") => true
+// DatabaseIncompatibleVersion("5.5.5-10.5.5-MariaDB") => false
+// DatabaseIncompatibleVersion("5.5.5-10.4.0-MariaDB") => true
+//
+// Adapted from link
+// https://stackoverflow.com/a/2941508
+bool DatabaseIncompatibleVersion(std::string const mysqlVersion) {
+    // Small anon func to turn a version string into an array of uint8
+    auto parse = [](const std::string& input)
+    {
+        std::vector<uint8> result;
+        std::istringstream parser(input);
+        result.push_back(parser.get());
+        for(int idx = 1; idx < 3; idx++)
+        {
+            parser.get(); //Skip period
+            result.push_back(parser.get());
+        }
+        return result;
+    };
+
+    // default to values for MySQL
+    uint8 offset = 0;
+    std::string minVersion = "5.7.0";
+
+    // If the version string contains "MariaDB", use that
+    if (mysqlVersion.find("MariaDB") != std::string::npos)
+    {
+        // All MariaDB 10.X versions have a prefix of 5.5.5 from the
+        // mysql_get_server_info() function
+        if (mysqlVersion.rfind("5.5.5-", 0) == 0)
+            offset = 6;
+        minVersion = "10.5.0";
```

Comment:
Instead of hardcoding minVersion here, can we move the definition to a header file so they are all stored together. Come the day we change requirements it will be a pain in the ass to find all the definition.

With that, also move the definition below to a headerfile alongside the defined above

https://github.com/azerothcore/azerothcore-wotlk/blob/7f7f18e48906959460171e05733baa736a730048/src/server/database/Database/DatabaseWorkerPool.cpp#L42-L48

Reply:
Yeah, that makes sense. I replaced the `MIN_MYSQL_SERVER_VERSION` since the spot I'm editing is the only place it's used